jack.py

```diff
@@ -600,6 +680,14 @@ def transport_locate(self, frame):
         _check(_lib.jack_transport_locate(self._ptr, frame),
                "Error locating JACK transport")
 
+    def transport_query(self):
+        """return current transport position."""
+        # Test rolling transport status
+        if _lib.jack_transport_query(self._ptr, self.position) == 1:
+            return self.position
+        else:
+            return None
```

Comment:
I make a new commit on my feature branch with class Transport. Now 
transport_query() method return a instance of class Transport. CFFI 
fields can be access with property or dict. dict is only created when 
transport_info() method is called.

About return value of _lib.jack_transport_query(), I put it in attribute 
'_state' of class Transport. Maybe I can just make '_state' attribute 
public : 'state' instead of creating a property (@property) ?

What do you think about my new commit ?

Julien

Le 14/04/2015 16:43, Matthias Geier a écrit :
>
> In jack.py 
> <https://github.com/spatialaudio/jackclient-python/pull/10#discussion_r28332847>:
>
> > @@ -600,6 +680,14 @@ def transport_locate(self, frame):
> >          _check(_lib.jack_transport_locate(self._ptr, frame),
> >                 "Error locating JACK transport")
> >
> > +    def transport_query(self):
> > +        """return current transport position."""
> > +        # Test rolling transport status
> > +        if _lib.jack_transport_query(self._ptr, self.position) == 1:
> > +            return self.position
> > +        else:
> > +            return None
>
>     Yes I think better is to return a tuple ?
>
> Yes, I think returning a tuple would be good.
>
> The question is if we can improve on the numbers in the enum 
> |jack_transport_state_t|.
> We can probably wrap this into a custom type that displays something 
> like |jack.ROLLING| ...
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/spatialaudio/jackclient-python/pull/10/files#r28332847>.
>